packages/logmein/src/index.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Append logmein=1 query parameter to mapped domain urls we want the user to be logged in against.
+ */
+
+// Used as placeholder / default domain to detect when we're looking at a relative url
+const INVALID_URL = `http://__domain__.invalid`;
+
+export function logmeinUrl( fullUrl: string ): string {
+	let url: URL;
+
+	try {
+		url = new URL( String( fullUrl ), INVALID_URL );
+	} catch ( e ) {
+		// Ignore unparseable urls
+		return fullUrl;
+	}
+
+	// Ignore and passthrough /relative/urls that have no host specified
+	if ( url.origin === INVALID_URL ) {
+		return fullUrl;
+	}
+
+	url.searchParams.set( 'logmein', '1' );
```

Comment:
Instead of appending logmein param, would it make sense to directly take a logged in user to the the r-login.wordpress.com endpoint? If the user is in Calypso and logged in, we don't need to take them to wordpress.com/log-in - we can take them to `https://r-login.wordpress.com/remote-login.php?action=link&back=https://{MAPPED_DOMAIN.tld}` and save the extra log in screen step:

<img width="1032" alt="Screenshot_2021-05-21_at_2_34_34_AM" src="https://user-images.githubusercontent.com/1269602/119057252-c47d5a00-b9e9-11eb-8245-ed04ea5f6762.png">

Appending logmein=1 is okay if we find the user is logged out for some reason. So, I'm thinking:

```
if ( isLoggedIn ) {
  navigate to https://r-login.wordpress.com/remote-login.php?action=link&back=https://{MAPPED_DOMAIN.tld}
} else {
  navigate to https://mapped-domain.tld/?logmein=1
}
```

A scenario in which this may not work is if the user has Calypso open in a tab, logs out of WPCOM in a different tab, and clicks on "Visit site" button on the open Calypso tab. Then they'll still be logged in to the mapped domain even though they're not signed in to WPCOM. Probably there are some checks we can add to handle this.

Reply:
This should be fine, I'll see what I can do.

Reply:
I tried this here: https://github.com/Automattic/wp-calypso/pull/52909/files#diff-d025c6a21229fe69836222916356ee8a2c395861252a82ff88e97b9facc70a4bR31

But it didn't seem to work for some reason (didn't dig too deep) end up getting js output as plaintext directly from `remote-login.php